src/App.jsx

```diff
@@ -7,22 +7,48 @@ HeadingH1.propTypes = {
   children: PropTypes.node.isRequired,
 };
 
-function useSemiPersistentState() {
-  const [todoList, setToDoList] = useState(() => {
-    const savedList = localStorage.getItem("savedTodoList");
+export default function App() {
+  const [todoList, setToDoList] = useState([]);
+  const [isLoading, setIsLoading] = useState(true);
+
+  // Mimicking a side-effect operation with a delay
+  function sideEffectHandler() {
+    return new Promise((resolve, reject) => {
+      setTimeout(() => {
+        const success = true;
 
-    return savedList ? JSON.parse(savedList) : [];
-  });
+        if (success) {
+          resolve();
```

Comment:
The `resolve` callback should be called passing the initial state for the todo list, which is that call to local storage.

The idea is to mimic a fetch call to the API, so the Promise must resolve with the response body like:
```json
{
  "data": [
    ...
  ]
}
```